src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+package com.fittracker.fittracker.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fittracker.fittracker.entity.User;
+import com.fittracker.fittracker.exception.ErrorResponse;
+import com.fittracker.fittracker.repository.UserRepository;
+import com.fittracker.fittracker.request.LoginRequest;
+import com.fittracker.fittracker.request.RegisterRequest;
+import com.fittracker.fittracker.response.RegisterResponse;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+import java.net.URI;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class AuthenticationControllerIntegrationTest extends BaseIntegrationTest{
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    private final static String ENDPOINT = "/auth";
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+
+    @ServiceConnection
+    static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:16");
+
+    @Override
+    protected List<HttpMethod> getProtectedHttpMethods() {
+        return List.of(POST);
```

Comment:
should be List.of(), AuthenticationController is the special case and the endpoints are not protected